frontend/src/containers/EditCollective.js

```diff
@@ -0,0 +1,370 @@
+  refpath: 'PublicGroupWhyJoin/PublicGroupWhyJoin-whyJoinText',
+  label: 'Update Text',
+  buttonClassName: 'EditButton--Text',
+  auxClassName: '-whyJoinText',
+  field: 'whyJoin',
+  extendStyle: function(target, rect, scrollY, style) {
+    const whyJoinMediaRect = this.refs['PublicGroupWhyJoin'].refs['PublicGroupWhyJoin-whyJoinMedia'].getBoundingClientRect();
+    style.left =  `${ whyJoinMediaRect.right }px`;
+    style.width = 'auto';
+  }
+}, {
+  refpath: 'PublicGroupWhyJoin/PublicGroupWhyJoin-whyJoinMedia',
+  label: 'Update Media',
+  buttonClassName: 'EditButton--Upload EditButton--left',
+  auxClassName: '-whyJoinMedia',
+  field: 'image',
+  extendStyle: function(target, rect, scrollY, style) {
+    // const whyJoinTextRect = this.refs['PublicGroupWhyJoin'].refs['PublicGroupWhyJoin-whyJoinText'].getBoundingClientRect();
+    style.right =  `${ rect.width + 70 }px`; // 70px is the padding on the edges of the layout.
+    style.width = 'auto';
+  }
+},
+].map(h => {
+  h.ref = `${h.refpath}--highlight`;
+  return h;
+});
+
+const Highlight = ({ ref, style, label, auxClassName, buttonClassName, onClick }) => (
+  <div ref={ ref } className='EditCollective-Highlight' style={ style } onClick={ onClick }>
+    <div className={`Highlight-aux ${ auxClassName }`}></div>
+    <div className={`EditCollective-EditButton ${ buttonClassName }`}>
+      <div className='EditCollective-EditButtonLabel'>{ label }</div>
+    </div>
+  </div>
+);
+
+const Viewport = props => (
+  <div className='EditCollective-Viewport'>
+    { props.children }
+    <div className='-screen'></div>
+  </div>
+);
+
+const Overlay = props => (
+  <div className='EditCollective-Overlay' onClick={ props.onClick }>
+    { props.children }
+  </div>
+);
+
+const Modal = props => (
+  <div className='EditCollective-Modal' onClick={e => {
+    e.nativeEvent.stopImmediatePropagation();
+    return false;
+  }}>
+    <div className='EditCollective-Modal-title'>
+      <span>{ props.title }</span>
+      <div className='-close' onClick={ props.onClose }>✖</div>
+    </div>
+    <div className='EditCollective-Modal-body'>
+      { props.children }
+      <div className='OnBoardingButton' onClick={ props.onClose }>Done</div>
+    </div>
+  </div>
+);
+
+const TopBar = props => (
+  <div className='EditCollective-TopBar'>
+    <div className='EditCollective-TopBar-brand'>
+      <svg width='18px' height='18px' className='-light-blue align-middle mr1'>
+        <use xlinkHref='#svg-isotype'/>
+      </svg>
+      <svg width='172px' height='30px' className='align-middle -logo-text'>
+        <use xlinkHref='#svg-logotype' fill='#fff' />
+      </svg>
+    </div>
+    <div className='EditCollective-TopBar-buttons'>
+      <div className='EditCollective-TopBar-Button' onClick={ props.onPublish }>Publish</div>
+      <a href='.'><div className='EditCollective-TopBar-Button trans'>Exit Edit Mode</div></a>
+    </div>
+  </div>
+);
+
+export default class EditCollective extends Component {
```

Comment:
In future please try to separate into presentational and container components (separate files). A component is either pure or performs side-effects such as sync with Redux, notify, hold a state, read globals, etc.

https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.kgdpj3cfy
https://medium.com/building-asana/designing-simpler-react-components-13a0061afd16#.mnvo638yp